app/assets/javascripts/Results/context_menu.js

```diff
@@ -0,0 +1,172 @@
+      * Depends on jQuery and jQuery-ui
+    * context_menu.css (..\app\assets\stylesheets\)
+
+  Context-Menu button functionality:
+    * _annotations.js.erb (..\app\views\results\common\)
+
+   */
+var annotation_context_menu = {
+  setup: function(annot_path, result_id, assignment_id, file_dl_path) {
+    var menu_items = {
+      new_annotation: {
+        title: 'New Annotation',
+        cmd: 'new_annotation',
+        action: make_new_annotation,
+        disabled: true
+      },
+      common_annotations: {
+        title: 'Common Annotations <kbd>></kbd>',
+        cmd: 'common_annotations',
+        disabled: true
+      },
+      delete_annotation: {
+        title: 'Delete Annotation',
+        cmd: "delete_annotation",
+        action: function(event, ui) {
+          var clicked_element = $(ui.target);
+          var annot_id = '';
+          var sub_file_id = $('#select_file_id').val();
+          if (annotation_type === ANNOTATION_TYPES.CODE) {
+            $.each(clicked_element[0].attributes, function(index, attr) {
+              if (attr.nodeName.toLowerCase()
+                      .indexOf('data-annotationid') != -1) {
+                annot_id = attr.value;
+                // Continue iteration in case of multiple annotations
+              }
+            });
+          } else {
+            annot_id = clicked_element.attr('id')
+                                      .replace('annotation_holder_', '');
+          }
+
+          if (annot_id !== null && annot_id.length !== 0 &&
+              sub_file_id !== null && sub_file_id.length !== 0) {
+            var query_str_params = [
+              ['id=' + annot_id],
+              ['submission_file_id=' + sub_file_id],
+              ['result_id=' + result_id],
+              ['assignment_id=' + assignment_id]
+            ];
+            $.ajax({
+              url: annot_path + '?' + query_str_params.join('&'),
+              method: 'POST',
+              data: { _method: 'delete' },
+              dataType: 'script'
+            });
+
+          }
+          return;
+        },
+        disabled: true
+      },
+      separator: {
+        title: '----'
+      },
+      copy: {
+        title: 'Copy',
+        cmd: 'copy',
+        action: function(){ document.execCommand('copy'); },
+        disabled: true
+      },
+      download: {
+        title: 'Download File',
+        cmd: 'download',
+        action: function() { download_func('false'); },
+        disabled: false
+      },
+      download_annotated: {
+        title: 'Download Annotated File',
+        cmd: 'download_annotated',
+        action: function() { download_func('true'); },
+        disabled: false
+      }
+    };
+
+    function download_func(include_annot) {
+      var sub_file_id = $('#select_file_id').val();
+      if (sub_file_id !== null && sub_file_id.length !== 0) {
+        window.open(file_dl_path + '?utf8=?&include_annotations=' +
```

Comment:
See my comment below about using the rails path helpers in this file. (You should be able to pass the query parameters as options to the helper as well.)